rdrf/report/reports/generator.py

```diff
@@ -197,141 +153,218 @@ def export_to_json(self, request):
-        pivoted = pivoted.droplevel('cfg.sortOrder', axis=1)
-
-        # Flatten the column levels into one, and reformat labels
-        pivoted.columns = pivoted.columns.to_series().str.join('_')
-        pivoted.columns = pivoted.columns.to_series().apply(self.__reformat_pivoted_column_labels)
-
-        # Remove null columns (caused by patients with no matching clinical data)
-        pivoted = pivoted.loc[:, pivoted.columns.notnull()]
-        # Remove duplicate columns (caused by repeated context form group's defaultName from how it's pivoted)
-        pivoted = pivoted.loc[:, ~pivoted.columns.duplicated()]
-
-        return stream_to_csv(pivoted)
+        def get_demographic_headers():
+            def get_flat_json_path(report_model, report_field, i=None):
+                if not report_field:
+                    return None
+                if report_model == 'patient':
+                    prefix = ''
+                else:
+                    prefix = f'{report_model}_'
+
+                field_json_path = re.sub(r"\s", "", report_field)
+                field_json_path = re.sub(r"(.+)({(.*)})", r"\1_\3", field_json_path)
+                if i is not None:
+                    return f"{prefix}{i}_{field_json_path}"
+                else:
+                    return f"{prefix}{field_json_path}"
+
+            def get_variants(lookup_key):
+                query = GqlQuery().fields([lookup_key]).query('dataSummary',
+                                                              input=self.patient_filters).operation().generate()
+                summary_result = self.schema.execute(query, context_value=request)
+                return summary_result.data['dataSummary'][lookup_key]
+
+            fieldnames_dict = OrderedDict()
+            fieldnames_dict['id'] = 'ID'
+
+            # e.g. {'patientAddress': 2}
+            processed_multifield_models = {}
+
+            for rdf in self.report_design.reportdemographicfield_set.all():
+                model_config = self.report_config[rdf.model]
+
+                if rdf.model == 'patient':
+                    # Get label for simple fields
+                    fieldnames_dict[get_flat_json_path(rdf.model, rdf.field)] = model_config['fields'][rdf.field]
+                else:
+                    if model_config.get('multi_field', False):
+                        if not processed_multifield_models.get(rdf.model):
+                            # Lookup how many variants of this model is relevant to our patient dataset
+                            num_variants = get_variants(model_config['variant_lookup'])
+
+                            # Process all the fields for this model now
+                            model_fields = self.report_design.reportdemographicfield_set.filter(
+                                model=rdf.model).values_list("field", flat=True)
+
+                            for i in range(0, (num_variants or 0)):
```

Comment:
The 0 is implied if you only provide the limit.

Reply:
Fixed